include/kodlab_mjbots_sdk/mjbots_control_loop.h

```diff
@@ -310,14 +318,14 @@ void MjbotsControlLoop<log_type, input_type, robot_type>::SafeProcessInput() {
   // Check to make sure using input
   if (input_) {
     // Try to unlock mutex, if you can't don't worry and try next time
-    if (lcm_sub_.mutex_.try_lock()) {
+    if (input_sub_.mutex_.try_lock()) {
       // If new message process
-      if (lcm_sub_.new_message_) {
+      if (input_sub_.new_message) {
         ProcessInput();
       }
       // Set new message to false and unlock mutex
-      lcm_sub_.new_message_ = false;
-      lcm_sub_.mutex_.unlock();
+      input_sub_.new_message = false;
+      input_sub_.mutex_.unlock();
     }
   }
 }
```

Comment:
Not an issue that needs to be changed in this PR but something to think about. 
We are handling safety of the input processing here and are using data as a public variable in the input_sub_ class. 
We will likely need a safe process input for every message or at least within any class that has an input message (behavior). Perhaps we could find a structure that makes data private and performs the mutex internally before returning a value copy or a ready flag, 

The solution doesnt have to match for behaviors and control loop, but it would be nice it was possible.

Reply:
Handled in ba8dcf1ad7d58d23fd9b58d79243d8c08754743c